src/transport/SessionManager.cpp

```diff
@@ -372,27 +374,49 @@ void SessionManager::ExpireAllPairingsForFabric(FabricIndex fabric)
     });
 }
 
+SessionHolder SessionManager::AllocateSession()
+{
+    SessionHolder holder;
+    Optional<SessionHandle> session = mSecureSessions.CreateNewSecureSession();
+    VerifyOrExit(session.HasValue(), holder = holder);
+    holder.Grab(session.Value());
+exit:
+    return holder;
+}
+
+SessionHolder SessionManager::AllocateSession(uint16_t sessionId)
+{
+    // If we forego SessionManager session ID allocation, we can have a
+    // collission.  In case of such a collission, we must evict first.
+    Optional<SessionHandle> oldSession = mSecureSessions.FindSecureSessionByLocalKey(sessionId);
+    if (oldSession.HasValue())
+    {
+        mSecureSessions.ReleaseSession(oldSession.Value()->AsSecureSession());
+    }
+    SessionHolder holder;
+    Optional<SessionHandle> session = mSecureSessions.CreateNewSecureSession(sessionId);
+    VerifyOrExit(session.HasValue(), holder = holder);
+    holder.Grab(session.Value());
+exit:
+    return holder;
+}
+
 CHIP_ERROR SessionManager::NewPairing(SessionHolder & sessionHolder, const Optional<Transport::PeerAddress> & peerAddr,
```

Comment:
With the `SessionHolder` retrievable from pairing session, I suggest removing `sessionHolder` argument here, The caller can get it via `pairing->GetSecureSessionHolder()`

Reply:
Would you be OK if I left it as is?  NewPairing was the one interface that survived this refactor.  While you're right that this is sort of redundant, it doesn't hurt anything.  And this delta is already pretty large.

Reply:
No problem, we can do it later